lib/_http_agent.js

```diff
@@ -123,6 +123,14 @@ Agent.prototype.addRequest = function(req, options) {
   options = util._extend({}, options);
   options = util._extend(options, this.options);
 
+  if (!options.servername) {
+    options.servername = options.host;
+    const hostHeader = req.getHeader('host');
+    if (hostHeader) {
+      options.servername = hostHeader.replace(/:.*$/, '');
```

Comment:
I'd prefer just using `hostHeader.slice()` and `hostHeader.indexOf()`, but if others are not comfortable with that, at the very least the regex should be `/:[\s\S]*$/` to match *any* character (including newlines).